controller/linodemachine_controller.go

```diff
@@ -473,6 +493,59 @@ func (r *LinodeMachineReconciler) resizeRootDisk(
 	return nil
 }
 
+func (r *LinodeMachineReconciler) UpdateInstanceConfigProfile(
+	ctx context.Context,
+	logger logr.Logger,
+	machineScope *scope.MachineScope,
+	linodeInstanceID int,
+) error {
+	// get the default instance config
+	configs, err := machineScope.LinodeClient.ListInstanceConfigs(ctx, linodeInstanceID, &linodego.ListOptions{})
+	if err != nil || len(configs) == 0 {
+		logger.Error(err, "Failed to list instance configs")
+
+		return err
+	}
+	instanceConfig := configs[0]
+
+	if machineScope.LinodeMachine.Spec.DataDisks != nil {
+		if err := r.createInstanceConfigDeviceMap(machineScope.LinodeMachine.Spec.DataDisks, instanceConfig.Devices); err != nil {
+			return err
+		}
+	}
+	if _, err := machineScope.LinodeClient.UpdateInstanceConfig(ctx, linodeInstanceID, instanceConfig.ID, linodego.InstanceConfigUpdateOptions{Devices: instanceConfig.Devices}); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r LinodeMachineReconciler) createInstanceConfigDeviceMap(instanceDisks map[string]*infrav1alpha1.InstanceDisk, instanceConfig *linodego.InstanceConfigDeviceMap) error {
+	for deviceName, disk := range instanceDisks {
+		switch deviceName {
+		case "sdb":
+			instanceConfig.SDB = ptr.To(linodego.InstanceConfigDevice{DiskID: disk.DiskID})
+		case "sdc":
+			instanceConfig.SDC = ptr.To(linodego.InstanceConfigDevice{DiskID: disk.DiskID})
+		case "sdd":
+			instanceConfig.SDD = ptr.To(linodego.InstanceConfigDevice{DiskID: disk.DiskID})
+		case "sde":
+			instanceConfig.SDE = ptr.To(linodego.InstanceConfigDevice{DiskID: disk.DiskID})
+		case "sdf":
+			instanceConfig.SDF = ptr.To(linodego.InstanceConfigDevice{DiskID: disk.DiskID})
+		case "sdg":
+			instanceConfig.SDG = ptr.To(linodego.InstanceConfigDevice{DiskID: disk.DiskID})
+		case "sdh":
+			instanceConfig.SDH = ptr.To(linodego.InstanceConfigDevice{DiskID: disk.DiskID})
+		default:
+			err := fmt.Errorf("unknown device name %s", deviceName)
+			return err
+		}
+	}
```

Comment:
nit: repetitive

```suggestion
	for deviceName, disk := range instanceDisks {
		dev := linodego.InstanceConfigDevice{
			DiskID: disk.DiskID,
		}
		switch deviceName {
		case "sdb":
			instanceConfig.SDB = &dev
		case "sdc":
			instanceConfig.SDC = &dev
		case "sdd":
			instanceConfig.SDD = &dev
		case "sde":
			instanceConfig.SDE = &dev
		case "sdf":
			instanceConfig.SDF = &dev
		case "sdg":
			instanceConfig.SDG = &dev
		case "sdh":
			instanceConfig.SDH = &dev
		default:
			return fmt.Errorf("unknown device name: %q", deviceName)
		}
	}
```

In the error message at the end, I suggest the use of the `%q` formatting verb, since it will wrap the string in quotes, which can be helpful if `deviceName == ""` (an empty string).